lib/rouge/lexers/varnish.rb

```diff
@@ -86,20 +102,23 @@ def self.variables
           push :inline_c
         end
 
-        rule %r/[a-z_.-]+/i do |m|
+        rule %r/\.?[a-z_][\w.-]*/i do |m|
           next token Keyword if self.class.keywords.include? m[0]
           next token Name::Function if self.class.functions.include? m[0]
           next token Name::Variable if self.class.variables.include? m[0]
           token Text
         end
 
         # duration
-        rule %r/(?:#{LNUM}|#{DNUM})(?:ms|[smhdwy])/, Num::Other
+        duration_suffix = Regexp.union(%w(ms s m h d w y))
+        rule %r/(?:#{INUM}|#{FNUM}|#{HINUM}|#{HFNUM})(?:#{duration_suffix})/, Num::Other
         # size in bytes
-        rule %r/#{LNUM}[KMGT]?B/, Num::Other
+        rule %r/#{INUM}[KMGT]?B/, Num::Other
         # literal numeric values (integer/float)
-        rule %r/#{LNUM}/, Num::Integer
-        rule %r/#{DNUM}/, Num::Float
+        rule %r/#{FNUM}/, Num::Float
+        rule %r/#{HFNUM}/, Num::Float
+        rule %r/#{INUM}/, Num::Integer
+        rule %r/#{HINUM}/, Num::Integer
```

Comment:
Is it possible to replace these with a generic regular expressions for numeric expressions and then do matches inside against the string matched? That would avoid needing to have so many local variables defined (each of which is using interpolation regular expressions). Those stick around in memory and are never garbage collected. Sometimes a couple are unavoidable but there are now a lot being added as part of this PR.

Reply:
Right. I've also refactored this group of token declarations to answer https://github.com/rouge-ruby/rouge/pull/1454#discussion_r389346888

faa2b37b630fda840a207a2c947cce392a98d900

See the core logic:

```ruby
       decimal = %r/[0-9]+/
        hex = %r/[0-9a-f]+/i

        numeric = %r{
          (?:
            0x#{hex}
            (?:\.#{hex})?
            (?:p[+-]?#{hex})?
          )
          |
          (?:
            #{decimal}
            (?:\.#{decimal})?
            (?:e[+-]?#{decimal})?
          )
        }xi
```

The `numeric` regexp says there are two kinds of numerics: hex, leading `0x`, and decimal.

Each kind has a required integer part, optional fraction part (e.g. ".123"), and optional exponential part (e.g. "e123"). Thus, "123", "123.456", "123.456e789", "123e456" are valid numeric, for example.

I also removed the "size in bytes" token in the commit, because I believe VCL has no such literals, at least in the latest Varnish including Fastly's.

Reply:
Got it. Thanks for the refactoring!